-1

**Failed Tests**: RelVals 
**Size**: This PR adds an extra 24KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-5e944a/52456/summary.html
**COMMIT**: 4e487863e12637058bde32cde4609d13e6f8e6b6
**CMSSW**: CMSSW_16_1_X_2026-04-02-2300/el8_amd64_gcc13
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/50653/52456/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Failed RelVals

- [13034.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-5e944a/52456/runTheMatrix-results/13034.0_TTbar_14TeV+2024PU)
```
----- Begin Fatal Exception 03-Apr-2026 13:25:20 CEST-----------------------
An exception of category 'FallbackFileOpenError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing module: class=MixingModule label='mix'
   [2] Calling RootInputFileSequence::initTheFile()
   [3] Calling StorageFactory::open()
   [4] Calling XrdFile::open()
Exception Message:
Failed to open the file 'root://xrootd-cms.infn.it//store/relval/CMSSW_14_2_0_pre4/RelValMinBias_14TeV/GEN-SIM/140X_mcRun3_2024_realistic_v26_STD_RegeneratedGS_2024_noPU-v1/2590000/869b7bf9-6b00-4309-b82a-2caba30c4c49.root?scitag.flow=196700'
   Additional Info:
      [a] Attempted to open logical file /store/relval/CMSSW_14_2_0_pre4/RelValMinBias_14TeV/GEN-SIM/140X_mcRun3_2024_realistic_v26_STD_RegeneratedGS_2024_noPU-v1/2590000/869b7bf9-6b00-4309-b82a-2caba30c4c49.root.
      [b] Failed to open the file with physical name root://eoscms.cern.ch//eos/cms/store/user/cmsbuild/store/relval/CMSSW_14_2_0_pre4/RelValMinBias_14TeV/GEN-SIM/140X_mcRun3_2024_realistic_v26_STD_RegeneratedGS_2024_noPU-v1/2590000/869b7bf9-6b00-4309-b82a-2caba30c4c49.root?scitag.flow=196700. Will attempt fallback.
      [c] Failed to open the file with physical name root://cms-xrd-global.cern.ch//eos/cms/store/relval/CMSSW_14_2_0_pre4/RelValMinBias_14TeV/GEN-SIM/140X_mcRun3_2024_realistic_v26_STD_RegeneratedGS_2024_noPU-v1/2590000/869b7bf9-6b00-4309-b82a-2caba30c4c49.root?scitag.flow=196700. Will attempt fallback.
      [d] Failed to open the file with physical name root://xrootd-cms.infn.it//store/relval/CMSSW_14_2_0_pre4/RelValMinBias_14TeV/GEN-SIM/140X_mcRun3_2024_realistic_v26_STD_RegeneratedGS_2024_noPU-v1/2590000/869b7bf9-6b00-4309-b82a-2caba30c4c49.root?scitag.flow=196700.
      [e] XrdCl::File::Open(name='root://xrootd-cms.infn.it//store/relval/CMSSW_14_2_0_pre4/RelValMinBias_14TeV/GEN-SIM/140X_mcRun3_2024_realistic_v26_STD_RegeneratedGS_2024_noPU-v1/2590000/869b7bf9-6b00-4309-b82a-2caba30c4c49.root?scitag.flow=196700', flags=0x10, permissions=0660) => error '[ERROR] Server responded with an error: [3011] No servers are available to read the file.
' (errno=3011, code=400). No additional data servers were found.
      [f] Last URL tried: root://cms-xrd-global.cern.ch:1094//store/relval/CMSSW_14_2_0_pre4/RelValMinBias_14TeV/GEN-SIM/140X_mcRun3_2024_realistic_v26_STD_RegeneratedGS_2024_noPU-v1/2590000/869b7bf9-6b00-4309-b82a-2caba30c4c49.root?scitag.flow=196700&tried=+1213xrootd-redic.pi.infn.it&xrdcl.requuid=f7c9d18c-66c8-45c6-8f09-62478ed030c1
      [g] Problematic data server: cms-xrd-global.cern.ch:1094
      [h] Disabled source: cms-xrd-global.cern.ch:1094
----- End Fatal Exception -------------------------------------------------
```

--- a/DQM/Integration/python/clients/ngt_dqm_sourceclient-live_cfg.py
+++ b/DQM/Integration/python/clients/ngt_dqm_sourceclient-live_cfg.py
@@ -89,6 +89,12 @@ process.dqmcommon = cms.Sequence(process.dqmEnv
 from PhysicsTools.Scouting.Run3ScoutingElectronBestTrackProducer_cfi import Run3ScoutingElectronBestTrackProducer as _Run3ScoutingElectronBestTrackProducer
 process.run3ScoutingElectronBestTrack =  _Run3ScoutingElectronBestTrackProducer.clone()
 
+# Metadata monitoring
+process.load("DQMOffline.Trigger.dqmHLTFiltersDQMonitor_cfi")
+process.dqmHLTFiltersDQMonitor.triggerEvent = 'hltTriggerSummaryAOD::HLT'
+process.dqmHLTFiltersDQMonitor.triggerResults = 'TriggerResults::HLT'
+process.dqmHLTFiltersDQMonitor.folderName = "NGT/Filters"
+
 process.p = cms.Path(process.dqmcommon *
                      process.hltOnlineBeamSpot *
                      process.run3ScoutingElectronBestTrack *
@@ -96,7 +102,8 @@ process.p = cms.Path(process.dqmcommon *
                      process.ScoutingRecHitsMonitoring *
                      process.ScoutingDileptonMonitorOnline *
                      process.ScoutingMuonPropertiesMonitorOnline *
-                     process.ScoutingPi0MonitorOnline)
+                     process.ScoutingPi0MonitorOnline *
+                     process.dqmHLTFiltersDQMonitor)
 
 ### process customizations included here
 from DQM.Integration.config.online_customizations_cfi import *

--- a/DQM/Integration/python/clients/ngt_dqm_sourceclient-live_cfg.py
+++ b/DQM/Integration/python/clients/ngt_dqm_sourceclient-live_cfg.py
@@ -94,6 +94,7 @@ process.load("DQMOffline.Trigger.dqmHLTFiltersDQMonitor_cfi")
 process.dqmHLTFiltersDQMonitor.triggerEvent = 'hltTriggerSummaryAOD::HLT'
 process.dqmHLTFiltersDQMonitor.triggerResults = 'TriggerResults::HLT'
 process.dqmHLTFiltersDQMonitor.folderName = "NGT/Filters"
+process.dqmHLTFiltersDQMonitor.lightMonitor = True
 
 process.p = cms.Path(process.dqmcommon *
                      process.hltOnlineBeamSpot *

--- a/DQMOffline/Trigger/plugins/HLTFiltersDQMonitor.cc
+++ b/DQMOffline/Trigger/plugins/HLTFiltersDQMonitor.cc
@@ -40,6 +40,7 @@ private:
   std::string processName_;
   bool initFailed_;
   bool skipRun_;
+  bool lightMonitor_;  // in case triggerEvent is missing from EventContent
 
   MonitorElement* meMenu_;
   std::unordered_map<std::string, MonitorElement*> meDatasetMap_;
@@ -61,6 +62,7 @@ HLTFiltersDQMonitor::HLTFiltersDQMonitor(const edm::ParameterSet& iConfig)
       processName_(""),
       initFailed_(false),
       skipRun_(false),
+      lightMonitor_(iConfig.getParameter<bool>("lightMonitor")),
       meMenu_(nullptr) {
   auto const& triggerResultsInputTag = iConfig.getParameter<edm::InputTag>("triggerResults");
 
@@ -172,6 +174,10 @@ void HLTFiltersDQMonitor::bookHistograms(DQMStore::IBooker& iBooker,
   for (size_t idx = 0; idx < triggerNames.size(); ++idx) {
     binIndexMap_[triggerNames[idx]] = idx + 1;
   }
+
+  // in case it's light monitoring, return here
+  if (lightMonitor_)
+    return;
 
   LogTrace("HLTFiltersDQMonitor") << "[HLTFiltersDQMonitor::bookHistograms] HLTConfigProvider::size() = "
                                   << hltConfigProvider_.size()
@@ -331,6 +337,10 @@ void HLTFiltersDQMonitor::analyze(const edm::Event& iEvent, const edm::EventSetu
                                            << iPathName << "\", bin_index=" << ibin;
     }
   }
+
+  // if it's light Monitoring do not do further work
+  if (lightMonitor_)
+    return;
 
   auto const& triggerEventHandle = iEvent.getHandle(triggerEventToken_);
   edm::Handle<trigger::TriggerEventWithRefs> triggerEventWithRefs;
@@ -536,6 +546,7 @@ void HLTFiltersDQMonitor::fillDescriptions(edm::ConfigurationDescriptions& descr
   edm::ParameterSetDescription desc;
   desc.add<std::string>("folderName", "HLT/Filters");
   desc.add<std::string>("efficPlotNamePrefix", "effic_");
+  desc.add<bool>("lightMonitor", false);
   desc.add<edm::InputTag>("triggerResults", edm::InputTag("TriggerResults::HLT"));
   desc.add<edm::InputTag>("triggerEvent", edm::InputTag("hltTriggerSummaryAOD::HLT"));
   desc.add<edm::InputTag>("triggerEventWithRefs", edm::InputTag("hltTriggerSummaryRAW::HLT"));

